include/micm/solver/cuda_lu_decomposition.hpp

```diff
@@ -0,0 +1,144 @@
+#pragma once
+#include<micm/solver/lu_decomposition.hpp>
+#include<micm/util/cuda_param.hpp>
+#include <stdexcept>
+#ifdef USE_CUDA
+#include <micm/solver/cuda_lu_decomposition.cuh>
+#endif 
+
+#ifdef USE_CUDA
+namespace micm{
+    class CUDALuDecomposition: public LuDecomposition{
+    public: 
+    CUDALuDecomposition(){}; 
+    
+    /// @brief Construct an LU decomposition algorithm for a given sparse matrix
+    /// @param matrix Sparse matrix
+    template<typename T, typename OrderingPolicy>
+    CUDALuDecomposition(const SparseMatrix<T, OrderingPolicy>& matrix); 
+    
+    template<typename T, template<class> typename SparseMatrixPolicy>
+    requires VectorizableSparse<SparseMatrixPolicy<T>>
+    void Decompose(
+        const SparseMatrixPolicy<T>&A, 
+        SparseMatrixPolicy<T>& L, 
+        SparseMatrixPolicy<T>& U) const; 
+    }; 
+
+    template<typename T, typename OrderingPolicy>
+    inline CUDALuDecomposition::CUDALuDecomposition(const SparseMatrix<T, OrderingPolicy>& matrix){
```

Comment:
Why is this function named `CUDAxxx`? It does not seem to run anything on GPU.

Reply:
This is the constructor of the derived class CUDALuDecomposition. Constructor in C++ has the same name as their class name.

Reply:
Thanks for the clarification. But since the whole function is not running on GPU, can we just remove it or move it to another class defined for CPU? Ideally anything in the `cuda_xxx.hpp` should be only GPU-related.

Reply:
I eliminated the constructor, and used the constructor from the base class.